src/services/SearchResults.cpp

```diff
@@ -230,21 +230,22 @@ void SearchResults::ProcessBlocksNibbles(const SearchResults& srSource, unsigned
     std::vector<unsigned int> vMatches;
     std::vector<unsigned char> vMemory;
     const unsigned int nPadding = Padding(m_nSize);
+    std::vector<unsigned char> pPrev;
 
     for (auto& block : srSource.m_vBlocks)
     {
         if (block.GetSize() > vMemory.capacity())
-            vMemory.reserve(block.GetSize());
+            vMemory.resize(block.GetSize());
 
-        unsigned char* pMemory = vMemory.data();
-        const unsigned char* pPrev = block.GetBytes();
+        for (auto i = 0U; i < block.GetSize(); i++)
+            pPrev.emplace_back(block.GetByte(i));
```

Comment:
You're still creating an unnecessary copy of the data. pPrev will be 4MB on an N64 game (8MB with the expansion pak).

Recomment replacing the "pPrev.at(i)" below with "block.GetByte(i)".

Reply:
done